src/controllers/transaction.controllers.js

```diff
@@ -0,0 +1,61 @@
+import transactionModel from '../models/transaction';
+import userControllers from './user.controllers';
+import { getById as getHouseById, changeOwner as changeHouseOwner } from './house.controllers';
+import { getByIdAbstract } from '../services/dbMethods';
+
+const addTransaction = async (buyerId, sellerId, houseId, price) => {
+  if (!buyerId || !sellerId || !houseId || !price) {
+    return;
+  }
+
+  const transaction = await transactionModel.create({
+    buyer: buyerId,
+    seller: sellerId,
+    house: houseId,
+    price,
+  });
+
+  return transaction;
+};
+
+export const buyHouse = async (data) => {
+  const { buyerId, houseId, price } = data;
+
+  let { data: buyer } = await userControllers.getById(buyerId);
+  let { data: house } = await getHouseById(houseId);
+  let { data: seller } = await userControllers.getById(house.owner);
```

Comment:
I'm getting `undefined` for `buyer` and `house` resulting in an error when it comes to this line. `getById` doesn't return anything and the destructurings of the results of the await calls don't expect an object with the structure `{ status: '', message: '' }`.